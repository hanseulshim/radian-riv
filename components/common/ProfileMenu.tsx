import { defaultAuth, useAuth } from 'context/auth/AuthProvider'
import Cookies from 'js-cookie'
import Link from 'next/link'
import React, { useCallback, useEffect, useRef, useState } from 'react'

export default function ProfileMenu() {
  const isCancelled = useRef(false)
  const [showMenu, setShowMenu] = useState(false)
  const { setAuth } = useAuth()

  const handleClick = useCallback(() => {
    if (!isCancelled.current) {
      setShowMenu(false)
    }
  }, [setShowMenu])

  useEffect(() => {
    if (showMenu) {
      document.addEventListener('click', handleClick)
    } else {
      document.removeEventListener('click', handleClick)
    }
  }, [showMenu])

  useEffect(() => {
    return () => {
      isCancelled.current = true
    }
  }, [])

  const logout = () => {
    Cookies.remove('auth')
    setAuth(defaultAuth)
  }
  const openMenu = () => {
    setShowMenu(true)
  }

  return (
    <div className="flex items-center dropdown-container">
      <img
        id="profile-logo"
        className="w-6 h-6 ml-2 cursor-pointer"
        alt="profile"
        src={`${process.env.baseUrl}/images/profile.svg`}
        onClick={openMenu}
      />
      {showMenu && (
        <ul className="dropdown -left-105" style={{ width: 150 }}>
          <Link href="/profile">
            <a>
              <li>Account</li>
            </a>
          </Link>
          <li>Help</li>
          <li onClick={logout}>Logout</li>
        </ul>
      )}
    </div>
  )
}
