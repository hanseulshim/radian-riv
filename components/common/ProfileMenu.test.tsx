import * as React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ProfileMenu from './ProfileMenu'

describe('ProfileMenu', () => {
  let component
  beforeAll(() => {
    component = shallow(<ProfileMenu />)
  })
  it('renders profile-menu', () => {
    expect(component.find('.profile-menu')).toHaveLength(1)
  })
  it('renders profile image', () => {
    expect(component.find('img#profile-logo').prop('src')).toContain(
      'profile.svg'
    )
  })
  it('should match snapshot before clicking image', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
  it('render menu container on click', () => {
    component.find('#profile-logo').simulate('click')
    expect(component.find('ul.menu-container')).toHaveLength(1)
    expect(component.find('li')).toHaveLength(3)
  })
  it('should match snapshot after clicking image', () => {
    expect(toJson(component)).toMatchSnapshot()
  })
})
