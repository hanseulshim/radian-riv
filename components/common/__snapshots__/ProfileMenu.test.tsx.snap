// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileMenu should match snapshot after clicking image 1`] = `
<div
  className="flex items-center dropdown-container"
>
  <img
    alt="profile"
    className="w-6 h-6 ml-2 cursor-pointer"
    id="profile-logo"
    onClick={[Function]}
    src="undefined/images/profile.svg"
  />
  <ul
    className="dropdown -left-105"
    style={
      Object {
        "width": 150,
      }
    }
  >
    <Link
      href="/profile"
    >
      <a>
        <li>
          Account
        </li>
      </a>
    </Link>
    <li>
      Help
    </li>
    <li
      onClick={[Function]}
    >
      Logout
    </li>
  </ul>
</div>
`;

exports[`ProfileMenu should match snapshot before clicking image 1`] = `
<div
  className="flex items-center dropdown-container"
>
  <img
    alt="profile"
    className="w-6 h-6 ml-2 cursor-pointer"
    id="profile-logo"
    onClick={[Function]}
    src="undefined/images/profile.svg"
  />
</div>
`;
